api/v1beta1/types_template.go

```diff
@@ -18,9 +18,289 @@ package v1beta1
 
 import (
 	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/utils/net"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
+// AzureManagedControlPlaneTemplateResourceSpec specifies an Azure managed control plane template resource.
+type AzureManagedControlPlaneTemplateResourceSpec struct {
```

Comment:
It seems that this is diverging from the AzureCluster ClassClass implementation in that it duplicates all the fields instead of creating an embedded "class" type that embeds the shared fields between AzureManagedControlPlane Spec and AzureManagedControlPlaneTemplate Spec https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/azurecluster_types.go#L35 

History on that design decision: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1943